ui/src/singletons/store.ts

```diff
@@ -84,156 +88,191 @@ interface AppStore {
+            }),
+
+          updateSelectedChildNode: (childId) =>
+            set(() => ({ selectedChildNode: childId })),
+
+          clearSelectedChildNode: () =>
+            set(() => ({ selectedChildNode: null })),
+
+          clearFlow: () =>
+            set(() => ({
+              nodes: [],
+              edges: [],
+              eipNodeConfigs: {},
+              selectedChildNode: null,
+            })),
+
+          clearDiagramSelections: () =>
+            set((state) => ({
+              nodes: state.nodes.map((node) => ({ ...node, selected: false })),
+              edges: state.edges.map((edge) => ({ ...edge, selected: false })),
+            })),
+
+          importFlowFromJson: (json: string) =>
+            set(() => {
+              const imported = JSON.parse(json) as Partial<AppStore>
+              if (isStoreType(imported)) {
+                return {
+                  nodes: imported.nodes,
+                  edges: imported.edges,
+                  eipNodeConfigs: imported.eipNodeConfigs,
+                }
+              }
+              console.error(
+                "Failed to import an EIP flow JSON. Malformed input"
+              )
+              return {}
+            }),
+
+          updateLayoutOrientation: (orientation: Layout["orientation"]) =>
+            set((state) => {
+              const newLayout: Layout = {
+                ...state.layout,
+                orientation: orientation,
+              }
+              const nodes = newFlowLayout(state.nodes, state.edges, newLayout)
+              return {
+                nodes: nodes,
+                layout: newLayout,
+              }
+            }),
+
+          updateLayoutDensity: () =>
+            set((state) => {
+              const newDensity =
+                state.layout.density === "compact" ? "comfortable" : "compact"
+              const newLayout: Layout = {
+                ...state.layout,
+                density: newDensity,
+              }
+              const nodes = newFlowLayout(state.nodes, state.edges, newLayout)
+              return {
+                nodes: nodes,
+                layout: newLayout,
+              }
+            }),
+        },
+      }),
+      {
+        limit: 20,
+
+        partialize: (state) => {
+          const newNodes = state.nodes.map((node) => {
+            const n = { ...node }
+            const {
+              width,
+              height,
+              selected,
+              draggable,
+              dragging,
+              positionAbsolute,
+              ...rest
+            } = n
+            return rest
+          })
+     
+          const { eipNodeConfigs, edges, layout} = state
+          return {eipNodeConfigs, layout, edges, nodes: newNodes}
+        },
+
+        equality: (pastState, currentState) =>  
+          isDeepEqual(pastState, currentState),
+
+        handleSet: (handleSet) =>
+          debounce((state) => handleSet(state), 1000, true),
```

Comment:
This is throwing a typescript compilation error (you can see it by running `npx tsc`), the following fixed it on my end:
```suggestion
          debounce<typeof handleSet>((state) => handleSet(state), 1000, true),
```